src/shogun/clustering/KMeansBase.cpp

```diff
@@ -134,6 +136,8 @@ void CKMeansBase::initialize_training(CFeatures* data)
 {
 	REQUIRE(distance, "Distance is not provided")
 	REQUIRE(distance->get_feature_type()==F_DREAL, "Distance's features type (%d) should be of type REAL (%d)")
+	REQUIRE(max_iter > 0, "The number of iterations must be greater than 0")
+	REQUIRE(k > 0, "The number of clusters must be greater than 0")
```

Comment:
in those messages, always print what was provided by the user. also missing newline

Reply:
:+1: